build.gradle

```diff
@@ -519,12 +520,32 @@ forbiddenApisTest.enabled = false
  * Maven Publish *
  *****************/
 
+void addManifest(Jar jar) {
+    jar.manifest {
+        attributes 'Implementation-Title': project.name
+        attributes 'Implementation-Vendor': 'HiveMQ'
+        attributes 'Implementation-Version': project.version
+    }
+}
+
+task embeddedJar(type: Jar) {
+    group 'build'
+
+    from sourceSets.main.output
+    addManifest(it)
+    exclude('META-INF/INDEX.LIST', 'META-INF/*.SF', 'META-INF/*.DSA', 'META-INF/*.RSA')
```

Comment:
why is this needed, we do not have any of these files

Reply:
you are correct, old habbit.